spread.yaml

```diff
@@ -66,6 +84,33 @@ prepare: |
     # this indicates that the server got reused, nothing to setup
     [ "$REUSE_PROJECT" != 1 ] || exit 0
 
+    # Disable burst limit so resetting the state quickly doesn't create problems.
+    mkdir -p /etc/systemd/system/snapd.service.d
+    cat <<EOF > /etc/systemd/system/snapd.service.d/local.conf
+    [Unit]
+    StartLimitInterval=0
+    [Service]
+    Environment=SNAPD_DEBUG_HTTP=7 SNAP_REEXEC=0
+    EOF
+
+    # adhoc backend and ubuntu-core system, nothing to setup
+    if [ "$SPREAD_BACKEND" = external-ubuntu-core ]; then
+        # build test binaries
+        if [ ! -f $GOPATH/bin/snapbuild ]; then
+            mkdir -p $GOPATH/bin
+            snap install --devmode --edge classic
+            classic "sudo apt update && apt install -y git golang-go"
+            classic "GOPATH=$GOPATH go get ..${PROJECT_PATH}/tests/lib/snapbuild"
+            snap remove classic
+        fi
+        mkdir -p /etc/systemd/system/snapd.socket.d
+        cat <<EOF > /etc/systemd/system/snapd.socket.d/local.conf
```

Comment:
Sorry, somehow this response got lost.
On amd64 images I was getting the same errors that we got with the service, now with the socket:
```
+ systemctl start snapd.service snapd.socket
A dependency job for snapd.service failed. See 'journalctl -xe' for details.
Job for snapd.socket failed. See "systemctl status snapd.socket" and "journalctl -xe" for details.
```
and in the journal:
```
Sep 19 16:07:05 localhost.localdomain systemd[1]: snapd.service: Job snapd.service/start failed with result 'dependency'.
Sep 19 16:07:05 localhost.localdomain systemd[1]: snapd.socket: Start request repeated too quickly.
Sep 19 16:07:05 localhost.localdomain systemd[1]: Failed to listen on Socket activation for snappy daemon.
-- Subject: Unit snapd.socket has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit snapd.socket has failed.
```
Setting the burst limit for the socket solves the issue, not sure if there's a better way.